daemon/logger/copier.go

```diff
@@ -40,14 +40,22 @@ func (c *Copier) Run() {
 
 func (c *Copier) copySrc(name string, src io.Reader) {
 	defer c.copyJobs.Done()
-	scanner := bufio.NewScanner(src)
-	for scanner.Scan() {
-		if err := c.dst.Log(&Message{ContainerID: c.cid, Line: scanner.Bytes(), Source: name, Timestamp: time.Now().UTC()}); err != nil {
-			logrus.Errorf("Failed to log msg %q for logger %s: %s", scanner.Bytes(), c.dst.Name(), err)
+	reader := bufio.NewReader(src)
+
+	for {
+		lineAndNewline, err := reader.ReadBytes('\n')
+		if err != nil {
+			if err == io.EOF {
+				return
+			}
+			logrus.Errorf("Error scanning log stream: %s", err)
+			return
```

Comment:
I have this sense that comparing `nil` with `io.EOF` at runtime causes a panic but I can't remember why... I'll give it a shot.